api/routes/states/data.js

```diff
@@ -0,0 +1,38 @@
+const defaultStateModel = require('../../db').models.state;
+const defaultUserModel = require('../../db').models.user;
+
+const getFields = [
+  'id',
+  'medicaid_office',
+  'name',
+  'program_benefits',
+  'program_vision',
+  'state_pocs'
+];
+const putFields = [
+  'medicaid_office',
+  'program_benefits',
+  'program_vision',
+  'state_pocs'
+];
+
+const getStateFromUserOrID = async (
+  stateID,
+  userID,
+  StateModel = defaultStateModel,
+  UserModel = defaultUserModel
+) => {
+  if (stateID) {
+    return StateModel.where({ id: stateID.toLowerCase() }).fetch();
+  }
+  const user = await UserModel.where({ id: userID }).fetch({
+    withRelated: ['state']
+  });
+  return user.related('state');
+};
+
+module.exports = {
+  getFields,
+  getStateFromUserOrID,
+  putFields
```

Comment:
These fields and this functionality are shared between the `GET` and `POST` endpoints.  I didn't want to muddy this PR too much by updating the GET endpoint to use them, but I will do that in a forthcoming PR.  That makes the tests easier, too - less database mocking!  :tada:

Reply:
nice